ingestion/src/metadata/utils/engines.py

```diff
@@ -16,23 +16,29 @@
 
 from sqlalchemy import create_engine
 from sqlalchemy.engine.base import Engine
+from sqlalchemy.exc import OperationalError
 from sqlalchemy.orm import sessionmaker
 from sqlalchemy.orm.session import Session
 
-from metadata.generated.schema.metadataIngestion.workflow import (
-    Source as WorkflowSource,
+from metadata.generated.schema.entity.services.connections.serviceConnection import (
+    ServiceConnection,
 )
 from metadata.utils.source_connections import get_connection_args, get_connection_url
 
 logger = logging.getLogger("Utils")
 
 
-def get_engine(workflow_source: WorkflowSource, verbose: bool = False) -> Engine:
+class SourceConnectionException(Exception):
+    """
+    Raised when we cannot connect to the source
+    """
+
+
+def get_engine(service_connection: ServiceConnection, verbose: bool = False) -> Engine:
```

Comment:
Updated get_engine to only require the ServiceConnection. We don't need the whole WorkflowSource here